x/nexus/types/types.go

```diff
@@ -30,9 +49,10 @@ func (m LinkedAddresses) Validate() error {
 	return nil
 }
 
-func (m ChainState) indexOfMaintainer(maintainer sdk.ValAddress) int {
-	for i, mt := range m.Maintainers {
-		if mt.Equals(maintainer) {
+// IndexOfMaintainer returns the index of the maintainer in the given chain state
+func (m ChainState) IndexOfMaintainer(address sdk.ValAddress) int {
```

Comment:
Why don't we use a stable map instead?

Reply:
Not quite sure if it is worth cuz it would make code elsewhere more complex. For example, we are looping through the chain maintainer states at https://github.com/axelarnetwork/axelar-core/blob/342871f593b464373fdf71d8b68db982f02aaa03/x/nexus/abci.go#L38. Unless the stable_marsharlier also makes sure looping through the map is deterministic, we would need to do more to ensure so.

Reply:
understood. Maps would still follow golangs definition, i.e. non-deterministic, so we'd have to sort it first. Keep it as is then